daffodil-io/src/main/scala/org/apache/daffodil/io/InputSourceDataInputStream.scala

```diff
@@ -129,23 +129,14 @@ final class InputSourceDataInputStream private (val inputSource: InputSource)
   def hasReachedEndOfData: Boolean = inputSource.hasReachedEndOfData
 
   /**
-   * The number of bits remaining if known. If a limit has not been set,
-   * returns Nope. If a limit has been set, the number of bits available
-   * until the limit as long as the end of data has not been seen. If the
-   * end of data has been seen, the smaller value of the remaining bits
-   * in the limit and the remaining bits in the data stream.
+   * Return the number of currently available bytes.
+   *
+   * This should not be used to determine the length of the data, as more bytes
+   * may become available in the future. This should really only be used for
+   * debug purposes.
```

Comment:
Might say "for debug or diagnostic purposes", just to make it clear it is safe to use in things like PEs.

Might also be worth mention that this does not take into account the bitLimit just so it's clear.

Reply:
Updated